sdk/scheduler/src/main/java/com/mesosphere/sdk/state/PersistentLaunchRecorder.java

```diff
@@ -35,56 +37,81 @@ public void record(OfferRecommendation offerRecommendation) throws Exception {
         LaunchOfferRecommendation launchOfferRecommendation = (LaunchOfferRecommendation) offerRecommendation;
         Protos.TaskInfo taskInfo = launchOfferRecommendation.getStoreableTaskInfo();
 
-        Protos.TaskStatus taskStatus = null;
+        Optional<PodInstance> podInstance = getPodInstance(taskInfo);
+        final boolean isInitialLaunch;
+        if (!podInstance.isPresent()) {
```

Comment:
This is doing a reverse lookup to find the ServiceSpec/PodSpec entry that a given TaskInfo is associated with.

It could be missing due to:
- TaskInfo without the expected labels (bad data, or extremely old data)
- ServiceSpec has been changed since this task was launched to no longer have this PodSpec

TODO: Put the above in a comment in the code